dev/io.openliberty.concurrent.internal.compat31/src/io/openliberty/concurrent/internal/compat/ContextServiceBase.java

```diff
@@ -0,0 +1,37 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.concurrent.internal.compat;
+
+import java.util.concurrent.Flow.Subscriber;
+
+import com.ibm.wsspi.threadcontext.ThreadContextDescriptor;
+
+import io.openliberty.concurrent.internal.compat.impl.ContextualSubscriber;
+
+/**
+ * Superclass for ContextServiceImpl when using Concurrency 3.0 or less.
```

Comment:
This should be Concurrency 3.1 or later, but can be changed in a later pull

Reply:
Good catch -- too much copying and pasting there!